pkg/generator/rps_test.go

```diff
@@ -0,0 +1,376 @@
+package generator
+
+import (
+	"github.com/vhive-serverless/loader/pkg/common"
+	"math"
+	"testing"
+)
+
+func TestWarmStartMatrix(t *testing.T) {
+	tests := []struct {
+		testName           string
+		experimentDuration int
+		rpsTarget          float64
+		expectedIAT        common.IATArray
+		expectedCount      []int
+	}{
+		{
+			testName:           "2min_1rps",
+			experimentDuration: 2,
+			rpsTarget:          1,
+			expectedIAT: []float64{
+				// minute 1
+				1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000, 1_000_000,
```

Comment:
This will be incompatible with the one generated with equidistant IAT in the trace mode: driver sleeps for the IAT in the minute and then invokes, and no invocation after last IAT in minute, since it is there just to align it to minute (or second) duration. Here, it won't generate expected number of invocations, but one less.

Reply:
I don't think this is relevant for RPS mode. The purpose of this mode is just to fire invocations for X minutes. The first and the last couple of minutes should discarded because of warmup and the issue you mention.